modules/sdk-coin-ada/src/lib/transactionBuilder.ts

```diff
@@ -255,6 +264,30 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
       outputs.add(changeOutput);
     }
 
+    // <DRAFT: support for multi-asset>
+    if (this._multiAssets !== undefined) {
+      let txOutputBuilder = CardanoWasm.TransactionOutputBuilder.new();
+      const toAddress = CardanoWasm.Address.from_bech32(this._transactionOutputs[0].address);
+      txOutputBuilder = txOutputBuilder.with_address(toAddress);
+      let txOutputAmountBuilder = txOutputBuilder.next();
+      const assetName = CardanoWasm.AssetName.new(Buffer.from(this._multiAssets.asset_name, 'hex'));
+      const policyId = CardanoWasm.ScriptHash.from_bytes(Buffer.from(this._multiAssets.policy_id, 'hex'));
+      const multiAsset = CardanoWasm.MultiAsset.new();
+      const assets = CardanoWasm.Assets.new();
+      assets.insert(assetName, CardanoWasm.BigNum.from_str(this._multiAssets.quantity));
+      multiAsset.insert(policyId, assets);
+
+      // should coin value be 0 since this output is related to token ?
+      const zeroValue = '0';
+      txOutputAmountBuilder = txOutputAmountBuilder.with_coin_and_asset(
+        CardanoWasm.BigNum.from_str(zeroValue),
+        multiAsset
+      );
+
+      const txOutput = txOutputAmountBuilder.build();
+      outputs.add(txOutput);
```

Comment:
Would `outputs.add(txOutput)` work if there were multiple outputs? Or is it already assuming there's multiple outputs?

Reply:
yes, had previously assumed a single assetList; just pushed the changes for multi asset list, which should add each output separately